### DISCUSSION TITLE
How to add a code to open the folder where the PDF is located?

### DISCUSSION THREAD

**Discussion by Sinoftj** [windingwind/zotero-actions-tags#436] [Q&A] [answered]
In the reader interface, it is not convenient to open the location of the file, which requires going back to the main interface.

**Comment by windingwind** [comment]
The shortcut and menu work in the reader. If not, it's because the action script isn't written properly. Please contact the author of the action script.

**Reply by Sinoftj** [reply]
打开PDF的所在文件夹，这个好像没有快捷键。所以我想问问有没有脚本可以实现的。

**Reply by ramonmi** [reply, marked as answer]
Event: None
Operation: Script
Shortcut: None
Name: Open PDF location

```
/**
 * @reference  https://github.com/windingwind/zotero-actions-tags/discussions/132
 * @usage Open the folder of the PDF attachment in Default Explorer
 * @link https://github.com/windingwind/zotero-actions-tags/discussions/436
 * @see https://github.com/windingwind/zotero-actions-tags/discussions/436
*/

if(!item) return;

const window = require("window");
const ZoteroPane = require("ZoteroPane");

// 如果是PDF，直接打开路径
if (item.isAttachment() && !item.isNote() && item.attachmentContentType === 'application/pdf') {
    ZoteroPane.showAttachmentInFilesystem(item.id);
    return "打开成功";
}
// 如果是父条目，查找其PDF附件并打开所在路径
else if (item.isRegularItem() && !item.isAttachment()) {
    let attachments = await item.getAttachments();
    for (let attachmentID of attachments) {
        let attachment = await Zotero.Items.getAsync(attachmentID);
        if (attachment.attachmentContentType === 'application/pdf') {
            ZoteroPane.showAttachmentInFilesystem(attachmentID);
	    return "打开成功";
        }
	//break;
    }
}
```

refs: https://github.com/windingwind/zotero-actions-tags/discussions/132

**Reply by Sinoftj** [reply]
呜呜呜 极其感谢

**Reply by mjthoraval** [reply]
Thank you very much for the script.
I have tried to assign the shortcut `Ctrl`+`o`. It works nicely from My Library.
It works also if I click on the tab of a PDF Reader tab.
But if I click inside the PDF Reader area, it does not work anymore.

Is the shortcut `Ctrl`+`o` already doing something else in a PDF Reader?

**Reply by ramonmi** [reply]
I don't know if this shortcut corresponds to other functions in the reader. But after enabling the debug output, you can see [Actions and Tags for Zotero] can't respond for the shortcut `Ctrl+o` in a reader (But it's different in the Collection/Library tab).
In Collection/Library tab
![image](https://github.com/user-attachments/assets/b3082044-f26d-41c7-8ac3-302db9cd8d27)

In reader tab
![image](https://github.com/user-attachments/assets/e232343e-e2dd-4914-b096-83897facac5b)

BTW, `ctrl+g`, `ctrl+s` also do not work in the reader.

So you may need to change to another shortcut.

**Reply by mjthoraval** [reply]
Thank you for checking.
Another question: can the code be modified to work also for other types of attachment files (snapshots, epubs, ...)?

**Reply by ramonmi** [reply]
Event: None
Operation: Script
Shortcut: None
Name: Open Attachment location
```js
/**
 * Open File Path
 * @reference  https://github.com/windingwind/zotero-actions-tags/discussions/132
 * @usage Open the folder of the attachment in the Default Explorer
 * @link https://github.com/windingwind/zotero-actions-tags/discussions/436
 * @see https://github.com/windingwind/zotero-actions-tags/discussions/436
*/

if(!item) return;

const ZoteroPane = require("ZoteroPane");

// Open the path directly for attachments
if (item.isAttachment() && !item.isNote()) {
    ZoteroPane.showAttachmentInFilesystem(item.id);
    return "[Open file path] Success!";
}
// Open the path of the best attachment for regular item
else if (item.isRegularItem() && !item.isAttachment()) {
    let attachments = await item.getAttachments();
    for (let attachmentID of attachments) {
	ZoteroPane.showAttachmentInFilesystem(attachmentID);
	return "[Open file path] Success!";
	//break;
    }
}
```

**Reply by mjthoraval** [reply]
Working perfectly. Thank you very much!

**Reply by ramonmi** [reply]
> Thank you for checking. Another question: can the code be modified to work also for other types of attachment files (snapshots, epubs, ...)?

The new script can open the path to any type of attachment. For your reference.
